api/jobs_test.go

```diff
@@ -311,14 +323,109 @@ func TestGetJobsHandler(t *testing.T) {
+				So(resp.Code, ShouldEqual, http.StatusOK)
+				payload, err := ioutil.ReadAll(resp.Body)
+				So(err, ShouldBeNil)
+				jobsReturned := models.Jobs{}
+				err = json.Unmarshal(payload, &jobsReturned)
+				So(err, ShouldBeNil)
+				zeroTime := time.Time{}.UTC()
+				expectedJob, err := ExpectedJob(validJobID2, zeroTime, 0, zeroTime, zeroTime, zeroTime, "Default Search Index Name", "created", 0, 0)
+				So(err, ShouldBeNil)
+
+				Convey("And the returned list should contain the expected job", func() {
+					returnedJobList := jobsReturned.JobList
+					So(returnedJobList, ShouldHaveLength, 1)
+					returnedJob := returnedJobList[0]
+					So(returnedJob.ID, ShouldEqual, expectedJob.ID)
+					So(returnedJob.Links, ShouldResemble, expectedJob.Links)
+					So(returnedJob.NumberOfTasks, ShouldEqual, expectedJob.NumberOfTasks)
+					So(returnedJob.ReindexCompleted, ShouldEqual, expectedJob.ReindexCompleted)
+					So(returnedJob.ReindexFailed, ShouldEqual, expectedJob.ReindexFailed)
+					So(returnedJob.ReindexStarted, ShouldEqual, expectedJob.ReindexStarted)
+					So(returnedJob.SearchIndexName, ShouldEqual, expectedJob.SearchIndexName)
+					So(returnedJob.State, ShouldEqual, expectedJob.State)
+					So(returnedJob.TotalSearchDocuments, ShouldEqual, expectedJob.TotalSearchDocuments)
+					So(returnedJob.TotalInsertedSearchDocuments, ShouldEqual, expectedJob.TotalInsertedSearchDocuments)
+				})
+			})
+		})
+
+		Convey("When a request is made to get a list of jobs with an offset greater than the total number of jobs in the Data Store", func() {
+			req := httptest.NewRequest("GET", "http://localhost:25700/jobs?offset=10&limit=20", nil)
+			resp := httptest.NewRecorder()
+
+			apiInstance.Router.ServeHTTP(resp, req)
+
+			Convey("Then a list of jobs is returned with status code 200", func() {
+				So(resp.Code, ShouldEqual, http.StatusOK)
+				payload, err := ioutil.ReadAll(resp.Body)
+				So(err, ShouldBeNil)
+				jobsReturned := models.Jobs{}
+				err = json.Unmarshal(payload, &jobsReturned)
+				So(err, ShouldBeNil)
+
+				Convey("And the returned list should be empty", func() {
+					returnedJobList := jobsReturned.JobList
+					So(returnedJobList, ShouldHaveLength, 0)
+				})
+			})
+		})
+
+		Convey("When a request is made to get a list of jobs with an offset that is not numeric", func() {
+			req := httptest.NewRequest("GET", "http://localhost:25700/jobs?offset=hi&limit=20", nil)
+			resp := httptest.NewRecorder()
+
+			apiInstance.Router.ServeHTTP(resp, req)
+
+			Convey("Then a bad request error is returned with status code 400", func() {
+				So(resp.Code, ShouldEqual, http.StatusBadRequest)
+				errMsg := strings.TrimSpace(resp.Body.String())
+				So(errMsg, ShouldEqual, expectedOffsetErrorMsg)
+			})
+		})
+
+		Convey("When a request is made to get a list of jobs with a limit that is negative", func() {
+			req := httptest.NewRequest("GET", "http://localhost:25700/jobs?offset=0&limit=-1", nil)
```

Comment:
Thanks for adding these tests 🙂  
For completeness, I would probably add cases for negative value in `offset` and non-numeric value in `limit`?

Reply:
no problem - it was quick to do